src/Arcus.Scripting.Bicep/Scripts/Inject-BicepContent.ps1

```diff
@@ -0,0 +1,126 @@
+    $fullPath = [System.IO.Path]::GetFullPath($combinedPath)
+    return $fullPath
+}
+
+function InjectFile {
+    param(
+        [string] $filePath
+    )
+
+    Write-Verbose "Checking Bicep template file '$filePath' for injection tokens..." 
+
+    $replaceContentDelegate = {
+        param($match)
+
+        $completeInjectionInstruction = $match.Groups[1].Value;
+        $instructionParts = @($completeInjectionInstruction -split "," | foreach { $_.Trim() } )
+
+        $filePart = $instructionParts[0];
+        # Regex uses non-capturing group for 'FileToInject' part, 
+        #  afterwards character classes and backreferencing to select the optional single or double quotes
+        $fileToInjectPathRegex = [regex] "^(?:FileToInject\s*=\s*)?([`"`']?)(?<File>.*?)\1?$";
+        $fileMatch = $fileToInjectPathRegex.Match($filePart)
+        if ($fileMatch.Success -ne $True){
+            throw "The file part '$filePart' of the injection instruction could not be parsed correctly"
+        }
+
+        $fullPathOfFileToInject = Get-FullyQualifiedChildFilePath -ParentFilePath $filePath -ChildFilePath $fileMatch.Groups["File"]
+        if (-not(Test-Path -Path $fullPathOfFileToInject -PathType Leaf)) {
+            throw "No file can be found at '$fullPathOfFileToInject'"
+        }
+
+        # Inject content recursively first
+        InjectFile($fullPathOfFileToInject)
+
+        Write-Verbose "`t Injecting content of '$fullPathOfFileToInject' into '$filePath'" 
+
+        $newString = Get-Content -Path $fullPathOfFileToInject -Raw
+
+        # XML declaration can only appear on the first line of an XML document, so remove when injecting
+        $newString = $newString -replace '(<\?xml).+(\?>)(\r)?(\n)?', ""
+
+        # By default: retain single quotes around content-to-inject, if present
+        $surroundContentWithSingleQuotes = $match.Value.StartsWith('''') -and $match.Value.EndsWith('''')
+
+        if ($instructionParts.Length -gt 1) {
+            $optionParts = $instructionParts | select -Skip 1
+
+            if ($optionParts.Contains("ReplaceSpecialChars")) {
+                Write-Host "`t Replacing special characters"
+
+                # Replace newline characters with literal equivalents
+                if ([Environment]::OSVersion.VersionString -like "*Windows*") {
+                    $newString = $newString -replace "`r`n", "\r\n"
+                } else {
+                    $newString = $newString -replace "`n", "\n"
+                }
+
+                # Replace tabs with spaces
+                $newString = $newString -replace "`t", "    "
+
+                # Replace ' with \'
+                $newString = $newString -replace "'", "\'"
+            }
+
+            if ($optionParts.Contains("InjectAsBicepObject")) {
+                $surroundContentWithSingleQuotes = $False
+            }
+        }
+
+        if ($surroundContentWithSingleQuotes) {
+            Write-Host "`t Surrounding content in double quotes"
```

Comment:
```suggestion
            Write-Verbose "`t Surrounding content in double quotes"
```
Since this is not controlled directly by user input, maybe made use `Write-Verbose`?
We may need to change this in the ARM-variant, too, I guess.

Reply:
Changed in [`6a44c09` (#357)](https://github.com/arcus-azure/arcus.scripting/pull/357/commits/6a44c095a8a0439593029593e6ad67ed924d398a)